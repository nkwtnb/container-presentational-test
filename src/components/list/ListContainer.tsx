import axios from "axios"
import { useEffect, useState } from "react";
import Presenter from "./Presenter";

export interface TodoType {
  userId: number
  id: number
  title: string
  completed: boolean
}

export interface PresenterProps {
  todos: TodoType[]
  handleClick: (() => void)
}

export default () => {
  const [todos, setTodos] = useState<TodoType[]>([]);
  const URL = "https://jsonplaceholder.typicode.com/todos/";

  useEffect(() => {
    (async () => {
      const resp = await axios.get(URL);
      setTodos(resp.data);
    })();
  }, []);
 
  return (
    <Presenter todos={todos}></Presenter>
  );
}