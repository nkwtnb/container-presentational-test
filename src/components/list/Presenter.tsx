import { TodoType } from "./ListContainer"

export default ({todos}: {todos: TodoType[]}) => {
  return (
    <div className="wrapper">
      {
        todos.length === 0
        ?
          <span data-testid="no-todos">データなし</span>
        :
          <div data-testid="todos">
          {
            todos.map((todo: any, index: number) => {
              return (
                <div key={index}>
                  {todo.title}
                </div>
              )
            })
          }
          </div>
      }
    </div>
  )
}