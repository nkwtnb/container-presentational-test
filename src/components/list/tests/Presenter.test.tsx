import React from 'react';
import { cleanup, prettyDOM, render, renderHook, RenderResult, screen, waitFor } from '@testing-library/react';
import ListContainer from '../ListContainer';
import Presenter from '../Presenter';
import { act } from 'react-dom/test-utils';

afterEach(() => {
  cleanup();
});

describe("Presenterのテスト", () => {
  test("todos = 0件 の時、「データなし」が表示されること", async () => {
    await act(() => {
    render(<Presenter todos={[]} />);
    })
    expect(screen.getByTestId("no-todos")).toHaveTextContent("データなし");
  })
  test("todos > 0件 の時、todo表示要素「todos」が表示されること", async () => {
    await act(() => {
      render(<Presenter todos={[{id: 1, title: "test", userId: 1, completed: true}]} />);
    })
    // state更新後の再描画を待機
    await waitFor(() => {
      expect(screen.getByTestId("todos")).toBeDefined();
    })
  });

  test("スナップショットが一致していること", async () => {
    let element: RenderResult;
    await act(() => {
      element = render(<ListContainer />);
    })
    expect(screen.getByTestId("no-todos")).toHaveTextContent("データなし");
    await waitFor(() => {
      expect(screen.getByTestId("todos").children.length).toBe(200);
    })
    // asFragment > toMatchSnapshotでスナップショットテスト
    expect(element!.asFragment()).toMatchSnapshot();
  });

});
