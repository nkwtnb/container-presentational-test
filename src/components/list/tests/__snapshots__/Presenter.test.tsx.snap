// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Presenterのテスト スナップショットが一致していること 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      data-testid="todos"
    >
      <div>
        delectus aut autem
      </div>
      <div>
        quis ut nam facilis et officia qui
      </div>
      <div>
        fugiat veniam minus
      </div>
      <div>
        et porro tempora
      </div>
      <div>
        laboriosam mollitia et enim quasi adipisci quia provident illum
      </div>
      <div>
        qui ullam ratione quibusdam voluptatem quia omnis
      </div>
      <div>
        illo expedita consequatur quia in
      </div>
      <div>
        quo adipisci enim quam ut ab
      </div>
      <div>
        molestiae perspiciatis ipsa
      </div>
      <div>
        illo est ratione doloremque quia maiores aut
      </div>
      <div>
        vero rerum temporibus dolor
      </div>
      <div>
        ipsa repellendus fugit nisi
      </div>
      <div>
        et doloremque nulla
      </div>
      <div>
        repellendus sunt dolores architecto voluptatum
      </div>
      <div>
        ab voluptatum amet voluptas
      </div>
      <div>
        accusamus eos facilis sint et aut voluptatem
      </div>
      <div>
        quo laboriosam deleniti aut qui
      </div>
      <div>
        dolorum est consequatur ea mollitia in culpa
      </div>
      <div>
        molestiae ipsa aut voluptatibus pariatur dolor nihil
      </div>
      <div>
        ullam nobis libero sapiente ad optio sint
      </div>
      <div>
        suscipit repellat esse quibusdam voluptatem incidunt
      </div>
      <div>
        distinctio vitae autem nihil ut molestias quo
      </div>
      <div>
        et itaque necessitatibus maxime molestiae qui quas velit
      </div>
      <div>
        adipisci non ad dicta qui amet quaerat doloribus ea
      </div>
      <div>
        voluptas quo tenetur perspiciatis explicabo natus
      </div>
      <div>
        aliquam aut quasi
      </div>
      <div>
        veritatis pariatur delectus
      </div>
      <div>
        nesciunt totam sit blanditiis sit
      </div>
      <div>
        laborum aut in quam
      </div>
      <div>
        nemo perspiciatis repellat ut dolor libero commodi blanditiis omnis
      </div>
      <div>
        repudiandae totam in est sint facere fuga
      </div>
      <div>
        earum doloribus ea doloremque quis
      </div>
      <div>
        sint sit aut vero
      </div>
      <div>
        porro aut necessitatibus eaque distinctio
      </div>
      <div>
        repellendus veritatis molestias dicta incidunt
      </div>
      <div>
        excepturi deleniti adipisci voluptatem et neque optio illum ad
      </div>
      <div>
        sunt cum tempora
      </div>
      <div>
        totam quia non
      </div>
      <div>
        doloremque quibusdam asperiores libero corrupti illum qui omnis
      </div>
      <div>
        totam atque quo nesciunt
      </div>
      <div>
        aliquid amet impedit consequatur aspernatur placeat eaque fugiat suscipit
      </div>
      <div>
        rerum perferendis error quia ut eveniet
      </div>
      <div>
        tempore ut sint quis recusandae
      </div>
      <div>
        cum debitis quis accusamus doloremque ipsa natus sapiente omnis
      </div>
      <div>
        velit soluta adipisci molestias reiciendis harum
      </div>
      <div>
        vel voluptatem repellat nihil placeat corporis
      </div>
      <div>
        nam qui rerum fugiat accusamus
      </div>
      <div>
        sit reprehenderit omnis quia
      </div>
      <div>
        ut necessitatibus aut maiores debitis officia blanditiis velit et
      </div>
      <div>
        cupiditate necessitatibus ullam aut quis dolor voluptate
      </div>
      <div>
        distinctio exercitationem ab doloribus
      </div>
      <div>
        nesciunt dolorum quis recusandae ad pariatur ratione
      </div>
      <div>
        qui labore est occaecati recusandae aliquid quam
      </div>
      <div>
        quis et est ut voluptate quam dolor
      </div>
      <div>
        voluptatum omnis minima qui occaecati provident nulla voluptatem ratione
      </div>
      <div>
        deleniti ea temporibus enim
      </div>
      <div>
        pariatur et magnam ea doloribus similique voluptatem rerum quia
      </div>
      <div>
        est dicta totam qui explicabo doloribus qui dignissimos
      </div>
      <div>
        perspiciatis velit id laborum placeat iusto et aliquam odio
      </div>
      <div>
        et sequi qui architecto ut adipisci
      </div>
      <div>
        odit optio omnis qui sunt
      </div>
      <div>
        et placeat et tempore aspernatur sint numquam
      </div>
      <div>
        doloremque aut dolores quidem fuga qui nulla
      </div>
      <div>
        voluptas consequatur qui ut quia magnam nemo esse
      </div>
      <div>
        fugiat pariatur ratione ut asperiores necessitatibus magni
      </div>
      <div>
        rerum eum molestias autem voluptatum sit optio
      </div>
      <div>
        quia voluptatibus voluptatem quos similique maiores repellat
      </div>
      <div>
        aut id perspiciatis voluptatem iusto
      </div>
      <div>
        doloribus sint dolorum ab adipisci itaque dignissimos aliquam suscipit
      </div>
      <div>
        ut sequi accusantium et mollitia delectus sunt
      </div>
      <div>
        aut velit saepe ullam
      </div>
      <div>
        praesentium facilis facere quis harum voluptatibus voluptatem eum
      </div>
      <div>
        sint amet quia totam corporis qui exercitationem commodi
      </div>
      <div>
        expedita tempore nobis eveniet laborum maiores
      </div>
      <div>
        occaecati adipisci est possimus totam
      </div>
      <div>
        sequi dolorem sed
      </div>
      <div>
        maiores aut nesciunt delectus exercitationem vel assumenda eligendi at
      </div>
      <div>
        reiciendis est magnam amet nemo iste recusandae impedit quaerat
      </div>
      <div>
        eum ipsa maxime ut
      </div>
      <div>
        tempore molestias dolores rerum sequi voluptates ipsum consequatur
      </div>
      <div>
        suscipit qui totam
      </div>
      <div>
        voluptates eum voluptas et dicta
      </div>
      <div>
        quidem at rerum quis ex aut sit quam
      </div>
      <div>
        sunt veritatis ut voluptate
      </div>
      <div>
        et quia ad iste a
      </div>
      <div>
        incidunt ut saepe autem
      </div>
      <div>
        laudantium quae eligendi consequatur quia et vero autem
      </div>
      <div>
        vitae aut excepturi laboriosam sint aliquam et et accusantium
      </div>
      <div>
        sequi ut omnis et
      </div>
      <div>
        molestiae nisi accusantium tenetur dolorem et
      </div>
      <div>
        nulla quis consequatur saepe qui id expedita
      </div>
      <div>
        in omnis laboriosam
      </div>
      <div>
        odio iure consequatur molestiae quibusdam necessitatibus quia sint
      </div>
      <div>
        facilis modi saepe mollitia
      </div>
      <div>
        vel nihil et molestiae iusto assumenda nemo quo ut
      </div>
      <div>
        nobis suscipit ducimus enim asperiores voluptas
      </div>
      <div>
        dolorum laboriosam eos qui iure aliquam
      </div>
      <div>
        debitis accusantium ut quo facilis nihil quis sapiente necessitatibus
      </div>
      <div>
        neque voluptates ratione
      </div>
      <div>
        excepturi a et neque qui expedita vel voluptate
      </div>
      <div>
        explicabo enim cumque porro aperiam occaecati minima
      </div>
      <div>
        sed ab consequatur
      </div>
      <div>
        non sunt delectus illo nulla tenetur enim omnis
      </div>
      <div>
        excepturi non laudantium quo
      </div>
      <div>
        totam quia dolorem et illum repellat voluptas optio
      </div>
      <div>
        ad illo quis voluptatem temporibus
      </div>
      <div>
        praesentium facilis omnis laudantium fugit ad iusto nihil nesciunt
      </div>
      <div>
        a eos eaque nihil et exercitationem incidunt delectus
      </div>
      <div>
        autem temporibus harum quisquam in culpa
      </div>
      <div>
        aut aut ea corporis
      </div>
      <div>
        magni accusantium labore et id quis provident
      </div>
      <div>
        consectetur impedit quisquam qui deserunt non rerum consequuntur eius
      </div>
      <div>
        quia atque aliquam sunt impedit voluptatum rerum assumenda nisi
      </div>
      <div>
        cupiditate quos possimus corporis quisquam exercitationem beatae
      </div>
      <div>
        sed et ea eum
      </div>
      <div>
        ipsa dolores vel facilis ut
      </div>
      <div>
        sequi quae est et qui qui eveniet asperiores
      </div>
      <div>
        quia modi consequatur vero fugiat
      </div>
      <div>
        corporis ducimus ea perspiciatis iste
      </div>
      <div>
        dolorem laboriosam vel voluptas et aliquam quasi
      </div>
      <div>
        inventore aut nihil minima laudantium hic qui omnis
      </div>
      <div>
        provident aut nobis culpa
      </div>
      <div>
        esse et quis iste est earum aut impedit
      </div>
      <div>
        qui consectetur id
      </div>
      <div>
        aut quasi autem iste tempore illum possimus
      </div>
      <div>
        ut asperiores perspiciatis veniam ipsum rerum saepe
      </div>
      <div>
        voluptatem libero consectetur rerum ut
      </div>
      <div>
        eius omnis est qui voluptatem autem
      </div>
      <div>
        rerum culpa quis harum
      </div>
      <div>
        nulla aliquid eveniet harum laborum libero alias ut unde
      </div>
      <div>
        qui ea incidunt quis
      </div>
      <div>
        qui molestiae voluptatibus velit iure harum quisquam
      </div>
      <div>
        et labore eos enim rerum consequatur sunt
      </div>
      <div>
        molestiae doloribus et laborum quod ea
      </div>
      <div>
        facere ipsa nam eum voluptates reiciendis vero qui
      </div>
      <div>
        asperiores illo tempora fuga sed ut quasi adipisci
      </div>
      <div>
        qui sit non
      </div>
      <div>
        placeat minima consequatur rem qui ut
      </div>
      <div>
        consequatur doloribus id possimus voluptas a voluptatem
      </div>
      <div>
        aut consectetur in blanditiis deserunt quia sed laboriosam
      </div>
      <div>
        explicabo consectetur debitis voluptates quas quae culpa rerum non
      </div>
      <div>
        maiores accusantium architecto necessitatibus reiciendis ea aut
      </div>
      <div>
        eum non recusandae cupiditate animi
      </div>
      <div>
        ut eum exercitationem sint
      </div>
      <div>
        beatae qui ullam incidunt voluptatem non nisi aliquam
      </div>
      <div>
        molestiae suscipit ratione nihil odio libero impedit vero totam
      </div>
      <div>
        eum itaque quod reprehenderit et facilis dolor autem ut
      </div>
      <div>
        esse quas et quo quasi exercitationem
      </div>
      <div>
        animi voluptas quod perferendis est
      </div>
      <div>
        eos amet tempore laudantium fugit a
      </div>
      <div>
        accusamus adipisci dicta qui quo ea explicabo sed vero
      </div>
      <div>
        odit eligendi recusandae doloremque cumque non
      </div>
      <div>
        ea aperiam consequatur qui repellat eos
      </div>
      <div>
        rerum non ex sapiente
      </div>
      <div>
        voluptatem nobis consequatur et assumenda magnam
      </div>
      <div>
        nam quia quia nulla repellat assumenda quibusdam sit nobis
      </div>
      <div>
        dolorem veniam quisquam deserunt repellendus
      </div>
      <div>
        debitis vitae delectus et harum accusamus aut deleniti a
      </div>
      <div>
        debitis adipisci quibusdam aliquam sed dolore ea praesentium nobis
      </div>
      <div>
        et praesentium aliquam est
      </div>
      <div>
        ex hic consequuntur earum omnis alias ut occaecati culpa
      </div>
      <div>
        omnis laboriosam molestias animi sunt dolore
      </div>
      <div>
        natus corrupti maxime laudantium et voluptatem laboriosam odit
      </div>
      <div>
        reprehenderit quos aut aut consequatur est sed
      </div>
      <div>
        fugiat perferendis sed aut quidem
      </div>
      <div>
        quos quo possimus suscipit minima ut
      </div>
      <div>
        et quis minus quo a asperiores molestiae
      </div>
      <div>
        recusandae quia qui sunt libero
      </div>
      <div>
        ea odio perferendis officiis
      </div>
      <div>
        quisquam aliquam quia doloribus aut
      </div>
      <div>
        fugiat aut voluptatibus corrupti deleniti velit iste odio
      </div>
      <div>
        et provident amet rerum consectetur et voluptatum
      </div>
      <div>
        harum ad aperiam quis
      </div>
      <div>
        similique aut quo
      </div>
      <div>
        laudantium eius officia perferendis provident perspiciatis asperiores
      </div>
      <div>
        magni soluta corrupti ut maiores rem quidem
      </div>
      <div>
        et placeat temporibus voluptas est tempora quos quibusdam
      </div>
      <div>
        nesciunt itaque commodi tempore
      </div>
      <div>
        omnis consequuntur cupiditate impedit itaque ipsam quo
      </div>
      <div>
        debitis nisi et dolorem repellat et
      </div>
      <div>
        ut cupiditate sequi aliquam fuga maiores
      </div>
      <div>
        inventore saepe cumque et aut illum enim
      </div>
      <div>
        omnis nulla eum aliquam distinctio
      </div>
      <div>
        molestias modi perferendis perspiciatis
      </div>
      <div>
        voluptates dignissimos sed doloribus animi quaerat aut
      </div>
      <div>
        explicabo odio est et
      </div>
      <div>
        consequuntur animi possimus
      </div>
      <div>
        vel non beatae est
      </div>
      <div>
        culpa eius et voluptatem et
      </div>
      <div>
        accusamus sint iusto et voluptatem exercitationem
      </div>
      <div>
        temporibus atque distinctio omnis eius impedit tempore molestias pariatur
      </div>
      <div>
        ut quas possimus exercitationem sint voluptates
      </div>
      <div>
        rerum debitis voluptatem qui eveniet tempora distinctio a
      </div>
      <div>
        sed ut vero sit molestiae
      </div>
      <div>
        rerum ex veniam mollitia voluptatibus pariatur
      </div>
      <div>
        consequuntur aut ut fugit similique
      </div>
      <div>
        dignissimos quo nobis earum saepe
      </div>
      <div>
        quis eius est sint explicabo
      </div>
      <div>
        numquam repellendus a magnam
      </div>
      <div>
        ipsam aperiam voluptates qui
      </div>
    </div>
  </div>
</DocumentFragment>
`;
